src/ocaml/typing/msupport_base.mli

```diff
@@ -0,0 +1,13 @@
+(** This file exists because Env need to be able to report errors, but Msupport has an
```

Comment:
Instead of this more complicated approach, I would just make a function reference exported from env.ml that gets set in Msupport. The mutable state is a bit gross, but it’s quite an effective and simple idiom, and I think easier to understand and maintain.